build/images/Dockerfile.build.windows

```diff
@@ -12,45 +12,39 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ARG WIN_BUILD_TAG
-ARG NANOSERVER_VERSION
-ARG WIN_BUILD_OVS_TAG
+ARG GO_VERSION
+ARG OVS_VERSION
 
-FROM antrea/base-windows:${WIN_BUILD_TAG} as antrea-build-windows
+FROM --platform=linux/amd64 golang:${GO_VERSION} as antrea-build-windows
+ARG CNI_BINARIES_VERSION
 
 WORKDIR /antrea
 
-COPY go.mod /antrea/go.mod
-
-RUN go mod download
+RUN --mount=type=cache,target=/go/pkg/mod/ \
+    --mount=type=bind,source=go.sum,target=go.sum \
+    --mount=type=bind,source=go.mod,target=go.mod \
+    go mod download
 
 COPY . /antrea
 
-RUN sh -c 'make windows-bin'
-
-FROM antrea/windows-ovs:${WIN_BUILD_OVS_TAG} as windows-ovs
-
-FROM mcr.microsoft.com/powershell:lts-nanoserver-${NANOSERVER_VERSION}
-SHELL ["pwsh", "-NoLogo", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
-
-LABEL maintainer="Antrea <projectantrea-dev@googlegroups.com>"
-LABEL description="A Docker image to deploy the Antrea CNI."
+RUN --mount=type=cache,target=/go/pkg/mod/ \
+    --mount=type=cache,target=/root/.cache/go-build/ \
+    make windows-bin
 
-USER ContainerAdministrator
+RUN mkdir -p /go/k/antrea/bin && mkdir -p /go/k/antrea/cni && \
+    cp /antrea/bin/antrea-agent.exe /go/k/antrea/bin/ && \
+    cp /antrea/bin/antctl.exe /go/k/antrea/bin/ && \
+    cp /antrea/build/images/scripts/Install-WindowsCNI.ps1 /go/k/antrea/ && \
+    cp /antrea/bin/antrea-cni.exe /go/k/antrea/cni/antrea.exe
 
-RUN mkdir -Force C:\k\antrea\bin
-COPY --from=antrea-build-windows  /opt/cni/bin /k/antrea/cni
-COPY --from=antrea-build-windows  /antrea/build/images/scripts/Install-WindowsCNI.ps1 /k/antrea/
-COPY --from=antrea-build-windows  /antrea/bin/antrea-agent.exe /k/antrea/bin/
-COPY --from=antrea-build-windows  /antrea/bin/antctl.exe /k/antrea/bin/antctl.exe
-COPY --from=antrea-build-windows  /antrea/bin/antrea-cni.exe /k/antrea/cni/antrea.exe
+ENV CNI_PLUGINS="./host-local.exe"
+ADD https://github.com/containernetworking/plugins/releases/download/${CNI_BINARIES_VERSION}/cni-plugins-windows-amd64-${CNI_BINARIES_VERSION}.tgz cni-plugins-windows-amd64.tgz
+RUN tar -xzf cni-plugins-windows-amd64.tgz -C /go/k/antrea/cni ${CNI_PLUGINS}
 
-RUN mkdir C:\k\antrea\utils
-COPY --from=antrea-build-windows /wins/wins.exe /k/antrea/utils/wins.exe
+FROM cattydong/windows-ovs:${OVS_VERSION} as antrea-ovs
```

Comment:
Is `cattydong` your personal registry?
Let me know if you want me to build the OVS base image locally and push it to the registry for you

Reply:
BTW, when I looked at the Docker documentation this other day, it wasn't clear that this was actually required. Does the following work:
```
COPY --from=cattydong/windows-ovs:${OVS_VERSION} /openvswitch /openvswitch
```

I think it should work but maybe it doesn't because of the image tag being a docker build argument?

Reply:
> Is cattydong your personal registry?

Yes, it is my personal registry, I use it to verify if the changes can pass CI verification as we don't have antrea/windows-ovs:3.0.5 yet.
 
It will be great if you can help push antrea/windows-ovs:3.0.5.

Reply:
> COPY --from=cattydong/windows-ovs:${OVS_VERSION} /openvswitch /openvswitch

I can have a try locally, and I will update if it works.

Reply:
It fails in my test, as what you suspect, either docker build argument or environment for the variable "OVS_VERSION" doesn't work when the variable is used in `from` section in a build stage. Below is the failure,

```
ERROR: failed to solve: failed to parse stage name "cattydong/windows-ovs:${OVS_VERSION}": invalid reference format
```

Reply:
@antoninbas I removed my personal registry in my latest change. Then github actions are failed, e.g.,`build-windows`, because of the missing dependency antrea/windows-ovs:3.0.5. Please notify me if you pushed the ovs image, then I can re-trigger the tests.

Reply:
thanks.